tapsend/allocation.go

```diff
@@ -505,6 +509,14 @@ func DistributeCoins(inputs []*proof.Proof, allocations []*Allocation,
 			},
 		)
 
+		// Before we create the virtual packet, we need to sort the
+		// proofs by amount and then by script key. This is important to
+		// ensure that the proofs are sorted in a deterministic way,
+		// before we assign them to the inputs of the virtual packet.
+		slices.SortFunc(proofsByID, func(i, j *proof.Proof) int {
```

Comment:
Not a comment, but just a confirmation that I'm understanding this PR.
This change (combined with the sorting in `wallet.go`) is the _actual_ fix right?

And you randomize the order of the assets in the unit test to check this is actually working.

:+1:

Reply:
Correct, this is the fix.
The change in the unit test shows that it fixes things, as we now need to re-order the expected inputs to be descending in their amounts (each tranche has a larger amount than the previous one, so with the new sort applied, the tranches with the higher number is expected to come first).